projects/js-packages/publicize-components/src/components/share-status/use-refresh-post-share-status.ts

```diff
@@ -0,0 +1,77 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { store as socialStore } from '../../social-store';
+import { ONE_MINUTE_IN_MS } from './constants';
+
+export type RefreshPostShareStatusOptions = {
+	postId?: number;
+	timeout?: number;
+};
+
+/**
+ * Refresh post share status.
+ *
+ * @param {RefreshPostShareStatusOptions} options - Options.
+ *
+ * @return {() => void} - Function to start polling.
+ */
+export function useRefreshPostShareStatus( {
+	postId,
+	timeout = ONE_MINUTE_IN_MS,
+}: RefreshPostShareStatusOptions = {} ) {
+	// @ts-expect-error `invalidateResolution` exists in every store
+	const { invalidateResolution } = useDispatch( socialStore );
+
+	const postShareStatus = useSelect(
+		select => select( socialStore ).getPostShareStatus( postId ),
+		[]
+	);
+
+	const [ hasTimeoutPassed, setHasTimeoutPassed ] = useState( false );
+	const [ pollForShareStatus, setPollForShareStatus ] = useState( false );
+	const [ lastTimestamp, setLastTimestamp ] = useState( 0 );
+
+	// If the last timestamp is provided, check if there are newer timestamps.
+	// If the last timestamp is not provided, check if there is done flag.
+	const isRequestComplete = lastTimestamp
+		? postShareStatus.shares.some( share => share.timestamp > lastTimestamp )
+		: postShareStatus.done;
```

Comment:
Can you tell me more about the uses of this done field? How is it set and does it get set even a post isn't shared?

Reply:
It's set [here](https://github.com/Automattic/jetpack/blob/b55642d5702f0a3d6d434800df8d21519b6dbaa0/projects/packages/publicize/src/class-rest-controller.php#L678-L679).

`done` is set to true when sharing for a post is processed, regardless of whether the post is actually shared or not, i.e. if `_publicize_shares` meta key exists, which can be empty array in case of post not being shared.

I created that flag before #39120 and I think we can get rid if it now and we can now rely upon `shares.length`